pkg/assembler/backends/neo4j/pkg.go

```diff
@@ -246,3 +253,210 @@ func (e *nameToVersion) PropertyNames() []string {
+						return nil, fmt.Errorf("string builder failed with err: %w", err)
+					}
+				}
+				queryValues["pkgType"] = pkgSpec.Type
+			}
+			if pkgSpec.Namespace != nil {
+
+				if firstMatch {
+					err := matchWhere(&sb, "namespace", "namespace", "$pkgNamespace")
+					if err != nil {
+						return nil, fmt.Errorf("string builder failed with err: %w", err)
+					}
+					firstMatch = false
+				} else {
+					err := matchAnd(&sb, "namespace", "namespace", "$pkgNamespace")
+					if err != nil {
+						return nil, fmt.Errorf("string builder failed with err: %w", err)
+					}
+				}
+				queryValues["pkgNamespace"] = pkgSpec.Namespace
+			}
+			if pkgSpec.Name != nil {
+
+				if firstMatch {
+					err := matchWhere(&sb, "name", "name", "$pkgName")
+					if err != nil {
+						return nil, fmt.Errorf("string builder failed with err: %w", err)
+					}
+					firstMatch = false
+				} else {
+					err := matchAnd(&sb, "name", "name", "$pkgName")
+					if err != nil {
+						return nil, fmt.Errorf("string builder failed with err: %w", err)
+					}
+				}
+				queryValues["pkgName"] = pkgSpec.Name
+			}
+			if pkgSpec.Version != nil {
+
+				if firstMatch {
+					err := matchWhere(&sb, "version", "version", "$pkgVerion")
+					if err != nil {
+						return nil, fmt.Errorf("string builder failed with err: %w", err)
+					}
+					firstMatch = false
+				} else {
+					err := matchAnd(&sb, "version", "version", "$pkgVerion")
+					if err != nil {
+						return nil, fmt.Errorf("string builder failed with err: %w", err)
+					}
+				}
+				queryValues["pkgVerion"] = pkgSpec.Version
+			}
+
+			if pkgSpec.Subpath != nil {
+
+				if firstMatch {
+					err := matchWhere(&sb, "version", "subpath", "$pkgSubpath")
+					if err != nil {
+						return nil, fmt.Errorf("string builder failed with err: %w", err)
+					}
+					firstMatch = false
+				} else {
+					err := matchAnd(&sb, "version", "subpath", "$pkgSubpath")
+					if err != nil {
+						return nil, fmt.Errorf("string builder failed with err: %w", err)
+					}
+				}
+				queryValues["pkgSubpath"] = pkgSpec.Subpath
+			}
+
+			if pkgSpec.MatchOnlyEmptyQualifiers != nil && !*pkgSpec.MatchOnlyEmptyQualifiers {
+
+				if len(pkgSpec.Qualifiers) > 0 {
+
+					for _, qualifier := range pkgSpec.Qualifiers {
+						// neo4j does not accept "." in its properties. If the qualifier contains a "." that must
+						// be replaced by an "-"
+						qualifierKey := strings.ReplaceAll(qualifier.Key, ".", "_")
+						if firstMatch {
+							err := matchWhere(&sb, "version", qualifierKey, "$"+qualifierKey)
+							if err != nil {
+								return nil, fmt.Errorf("string builder failed with err: %w", err)
+							}
+							firstMatch = false
+						} else {
+							err := matchAnd(&sb, "version", qualifierKey, "$"+qualifierKey)
+							if err != nil {
+								return nil, fmt.Errorf("string builder failed with err: %w", err)
+							}
+						}
+						queryValues[qualifierKey] = qualifier.Value
+					}
+				}
+			}
```

Comment:
should there be an else here where we want to match only on empty qualifiers?

Reply:
I think it should be covered by `MatchOnlyEmptyQualifiers`

Reply:
i think the code here just omits the qualifiers, but doesn't look for nodes with no qualifiers (filtering on no qualifiers), don't think there's a good way to query this right now. Unless i'm understanding the flag incorrectly.

Reply:
@mihaimaruseac and I discussed this. We have a path forward to remedy this.

Reply:
the method we discussed might not work. Looking at alternatives.